### PR TITLE
Support for Parsing with Jackson.

### PR DESCRIPTION
**Pull request by lbwexler**
Also includes enhancement to JSONClient to set default headers.

See ticket for more details:
https://github.com/xh/hoist-core/issues/128

Please also merge the following:
https://github.com/xh/toolbox/pull/394

### CODE REVIEW

File: src/main/groovy/io/xh/hoist/http/JSONClient.groovy [hunk 1]
```diff
@@ -88,6 +125,14 @@ class JSONClient {
         String statusText
         boolean success
 
+        if (!method.containsHeader('Content-type') && method.method in ['POST', 'PUT']) {
```
Comment (lbwexler) [inline src/main/groovy/io/xh/hoist/http/JSONClient.groovy:128]: Patch?  Look at body property?

File: src/main/groovy/io/xh/hoist/json/JSON.groovy [hunk 2]
```diff
@@ -10,13 +10,15 @@ package io.xh.hoist.json
 import groovy.transform.CompileStatic
 
 /**
- * Alias for Grails JSON converter. Used for parsing JSON strings to Objects in Groovy.
+ * Alias for Grails JSON converter.
  *
- * In prior versions of this plugin, this class also supported customized serialization from Objects
- * to JSON. Applications should now use the Jackson-based JSONSerializer instead when writing JSON.
+ * DEPRECATED. This class is maintained here primarily for backwards compatibility; no future
+ * extensions or modifications are currently planned. Applications should consider using either
```
Comment (lbwexler) [inline src/main/groovy/io/xh/hoist/json/JSON.groovy:16]: grammar

**Comment by lbwexler** [comment]
@amcclain -- think this is ready to merge.  Do you want to take another look with the API changes?  If looks good to you, please merge

**Comment by haynesjm42** [comment]
👍

**Pull request merged**